app/api/api/views.py

```diff
@@ -525,6 +534,51 @@ def perform_create(self, serializer, non_file_serializer):
         add_message(os.environ.get("UPLOAD_QUEUE_NAME"), azure_dict)
 
 
+class StructuralMappingTableAPIView(APIView):
+    def post(self, request, *args, **kwargs):
+        try:
+            body = request.data
+        except ValueError:
+            body = {}
+        if (
+            request.POST.get("download_rules") is not None
+            or body.get("download_rules", None) is not None
+        ):
+            qs = self.get_queryset()
+            return download_mapping_rules(request, qs)
+        elif (
+            request.POST.get("download_rules_as_csv") is not None
+            or body.get("download_rules_as_csv", None) is not None
+        ):
+            qs = self.get_queryset()
+            return download_mapping_rules_as_csv(request, qs)
+        elif (
+            request.POST.get("get_svg") is not None
+            or body.get("get_svg", None) is not None
+        ):
+            qs = self.get_queryset()
+            return view_mapping_rules(request, qs)
+        else:
+            return Response(
+                {"error": "not working right now!"}, status=status.HTTP_400_BAD_REQUEST
```

Comment:
This message implies a server error, but you're returning Bad Request. I'd change either the message or the code.